aten/src/THC/generic/THCTensorMath.cu

```diff
@@ -396,9 +396,11 @@ accreal THCTensor_(trace)(THCState *state, THCTensor *src_) {
 
 void THCTensor_(linspace)(THCState *state, THCTensor *r_, real a, real b, int64_t n) {
   THCAssertSameGPU(THCTensor_(checkGPU)(state, 1, r_));
-  THArgCheck(n > 1 || (n == 1 && (a == b)), 3, "invalid number of points");
+  // NumPy allows you to pass different points even if n <= 1 -- should we?
+  THArgCheck(n > 1 || ((n == 0 || n == 1) && (a == b)), 3, "invalid number of points");
   if (THCTensor_(nElement)(state, r_) != n) THCTensor_(resize1d)(state, r_, n);
-  if (n == 1) THCTensor_(fill)(state, r_, a);
+  if (n == 0) {
+  } else if (n == 1) THCTensor_(fill)(state, r_, a);
```

Comment:
I found this a little hard to read. Empty if statements are awkward so I usually put a comment to show that it's intentional. Something like:

```
if (n == 0) {
 // skip
} else if (n == 1) {
 THCTensor_(fill)(state, r_, a);
} else {
  ...
}
```

Reply:
sure, good suggestion.